azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java

```diff
@@ -0,0 +1,273 @@
+            "(?:           # Range for 1st value (optional).    \n" +
+            "  -           # Dash separates range integer.      \n" +
+            "  [0-9]+      # Range integer of 1st value.        \n" +
+            ")?            # Range for 1st value (optional).    \n" +
+            "(?:           # Zero or more additional values.    \n" +
+            "  ,           # Comma separates additional values. \n" +
+            "  [0-9]+      # Integer of extra value (required). \n" +
+            "  (?:         # Range for extra value (optional).  \n" +
+            "    -         # Dash separates range integer.      \n" +
+            "    [0-9]+    # Range integer of extra value.      \n" +
+            "  )?          # Range for extra value (optional).  \n" +
+            ")*            # Zero or more additional values.    \n" +
+            "$             # Anchor to end of string.           ", Pattern.COMMENTS);
+
+    private static final Pattern REGEX_NEXT_INTEGER_RANGE = Pattern.compile("# extract next integers/integer range value.    \n" +
+            "([0-9]+)      # $1: 1st integer (Base).         \n" +
+            "(?:           # Range for value (optional).     \n" +
+            "  -           # Dash separates range integer.   \n" +
+            "  ([0-9]+)    # $2: 2nd integer (Range)         \n" +
+            ")?            # Range for value (optional). \n" +
+            "(?:,|$)       # End on comma or string end.", Pattern.COMMENTS);
+
+    private BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public String promoteString(String name, String defaultValue, String regex, boolean isRequired) throws IOException {
+        final boolean hasDefaultValue = StringUtils.isNotBlank(defaultValue);
+        final String hintMessage = hasDefaultValue ? (" (" + TextUtils.green(defaultValue) + ")") : "";
+        System.out.printf("Please input the %s%s:", name, hintMessage);
+        System.out.flush();
+
+        while (true) {
+
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
+                if (hasDefaultValue || !isRequired) {
+                    return defaultValue;
+                }
+            }
+            if (Pattern.compile(regex).matcher(input).find()) {
+                return input;
+            } else {
+                System.out.printf(PROMOT_MESSAGE_STRING_TEMPLATE, input, name, hintMessage);
+                System.out.flush();
+            }
+        }
+    }
+
+    public Boolean promoteYesNo(Boolean defaultValue, String message) throws IOException {
+        final String hintMessage = yesOrNo(defaultValue);
+        System.out.printf("%s %s: ", message, hintMessage);
+        System.out.flush();
+        while (true) {
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
+                return defaultValue;
```

Comment:
defaultValue == null ?

Reply:
press enter will keep original value or default value, if there is no original and default value, return null means get nothing.